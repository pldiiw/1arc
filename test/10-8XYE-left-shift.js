'use strict';

let engine = require('engine').initialize();
const leftShift = require('instructions').leftShift;

engine.data[1] = 121;
engine.data[0xA] = 200;

const engine1 = leftShift(engine, 1, 1);
const engine2 = leftShift(engine1, 1, 0xA);

console.log(engine.data[1]);
console.log(engine.data[0xA]);
console.log(engine.data[0xF]);

console.log(engine1.data[1]);
console.log(engine1.data[0xA]);
console.log(engine1.data[0xF]);

console.log(engine2.data[1]);
console.log(engine2.data[0xA]);
console.log(engine2.data[0xF]);

