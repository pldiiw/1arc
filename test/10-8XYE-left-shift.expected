121
200
0
242
200
0
144
200
1
